src/adapters/LayerZeroAdapter.sol

```diff
@@ -70,11 +72,14 @@ contract LayerZeroAdapter is Auth, ILayerZeroAdapter {
         external
         payable
     {
+        require(payload.toUint64(0) >= gasleft());
```

Comment:
Maybe we should add some offset for the current waste gas:

require(payload.toUint64(0) >= gasleft() - OFFSET);